feat: APP-529 receipt modal for crypto purchases

r41ph opened the pull request:
## Description

https://regennetwork.atlassian.net/browse/APP-529

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. Go to https://deploy-preview-2586--regen-marketplace.netlify.app/
2. Login and go to My Orders
3. Verify that Orders using crypto have a receipt button that opens a modal window with the receipt information

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

r41ph commented:
@erikalogie @S4mmyb see testing instructions.

S4mmyb commented:
@r41ph I'm not seeing the my orders dropdown in the menu

blushi commented:
> @r41ph I'm not seeing the my orders dropdown in the menu

Have you made any purchase with this account? If not, then it's hidden.

blushi commented on web-marketplace/src/components/organisms/OrderCryptoReceiptModal/OrderCryptoReceiptModal.tsx:
```diff
@@ -0,0 +1,115 @@
+import { Box } from '@mui/material';
+import { USD_DENOM } from 'config/allowedBaseDenoms';
+
+// import OutlinedButton from 'web-components/src/components/buttons/OutlinedButton';
```
can we clean that up?

r41ph replied:
We will be using it soon when we implement the 'Print Receipt' button that is also commented out at the bottom of the file.

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -29,40 +26,29 @@ interface OrderSummaryProps {
-      }),
-    [allowedDenoms, askBaseDenom, askDenom],
-  );
+  const { credits } = creditsData;
```
maybe we could just pass in `credits={creditsData.credits}` as prop instead of the full `creditsData` since we do not use the other properties here directly

blushi commented on web-marketplace/src/components/organisms/Order/Order.tsx:
```diff
@@ -37,107 +41,178 @@ export const Order = ({ orderData, allowedDenoms, className }: OrderProps) => {
+            },
+            amount: credits.credits,
+            date: blockchainDetails.purchaseDate,
+            txType: cryptoReceiptModalState.type,
```
not sure if `txType` is the right term here, it's not really about the tx but rather about the credits being retired or not
also we could simply provide `retirementInfo.retiredCredits` instead of getting it from `cryptoReceiptModalState`, and then based on `retiredCredits` bool value we display the appropriate content in `OrderCryptoReceiptModal`

blushi reviewed (commented):
Looks good overall, just a few comments

![image](https://github.com/user-attachments/assets/d4a5f7ac-269d-4a43-83c4-742710ce726d)

We should probably have a larger space between the buttons and the left part of the card with the project name, location and size.

erikalogie commented:
LGTM, just some small spacing issue:
<img width="656" alt="Monosnap Regen Marketplace : Buy, trade and retire carbon credits | Regen Marketplace : Buy, trade and retire carbon credits 2025-01-28 07-30-39" src="https://github.com/user-attachments/assets/b2960cac-9658-44c2-9eb1-dcedc1421c01" />

S4mmyb commented:
The tooltip question-mark looks a little off center compared to the comps (i.e. the question mark logo doesn't look centered). Also in terms of of the placement maybe we could move it a little down and left so it doesn't look like it's hanging as much? I'll let @erikalogie chime in with thoughts though. 

![image](https://github.com/user-attachments/assets/6038999e-4e30-475b-b244-a966adf86a23)

r41ph commented:
> We should probably have a larger space between the buttons and the left part of the card with the project name, location and size.

I already created an issue for this and for the way buttons align in smaller screens: https://regennetwork.atlassian.net/browse/APP-556 - Shall I work on it here @erikalogie @blushi ?

blushi commented:
> > We should probably have a larger space between the buttons and the left part of the card with the project name, location and size.
> 
> I already created an issue for this and for the way buttons align in smaller screens: https://regennetwork.atlassian.net/browse/APP-556 - Shall I work on it here @erikalogie @blushi ?

Cool, let's work on this separately then!

erikalogie commented:
> The tooltip question-mark looks a little off center compared to the comps (i.e. the question mark logo doesn't look centered). Also in terms of of the placement maybe we could move it a little down and left so it doesn't look like it's hanging as much? I'll let @erikalogie chime in with thoughts though.
> 
> ![image](https://private-user-images.githubusercontent.com/67330492/407497419-6038999e-4e30-475b-b244-a966adf86a23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxNTg4MTQsIm5iZiI6MTczODE1ODUxNCwicGF0aCI6Ii82NzMzMDQ5Mi80MDc0OTc0MTktNjAzODk5OWUtNGUzMC00NzViLWIyNDQtYTk2NmFkZjg2YTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI5VDEzNDgzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1ODRhMGVhMDU4MjlhYzdiNTllZmFmNjJhMjZlNzZkNjYxZjJjNjE5NjZmNmMzOTA1OTU2MDJkZWNkZjg5OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xBhaQWvBj5kpDl3k1pt7PIfVhg5EbfFQGUGU8NjGIKQ)

Yeah good catch on both of these

r41ph commented:
@S4mmyb I have added the tooltip issue you raised to the ticket I mentioned [here](https://github.com/regen-network/regen-web/pull/2586#issuecomment-2621052067), where we are going to work on a couple more issues in the My Orders page.

@erikalogie I have updated the spacing as per your comment. See image below because as discussed yesterday in the Core design meeting this feature is going to not be available for now, so the preview doesn't have the 'View receipt' button for crypto visible now. 

![image](https://github.com/user-attachments/assets/01f71568-d56e-4891-8f73-3d8243c1e1b1)

erikalogie commented:
LGTM
